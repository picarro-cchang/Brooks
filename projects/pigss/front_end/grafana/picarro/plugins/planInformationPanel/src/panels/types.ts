import { any } from "prop-types";

export interface Plan {
  max_steps: number;
  panel_to_show: number;
  current_step: number;
  focus: PlanFocus;
  last_step: number;
  steps: { [key: string]: PlanStep };
  num_plan_files: number;
  plan_files: { [key: string]: string };
  plan_filename: string;
  bank_names: {
    [key: number]: {
      name: string;
      channels: { [key: number]: string };
    };
  };
}

export interface PlanFocus {
  row: number;
  column: number;
}

export interface BankConfig {
  chan_mask: number;
  clean: number;
}

export interface PlanStep {
  banks: { [key: number]: BankConfig };
  reference: number;
  duration: number;
}
