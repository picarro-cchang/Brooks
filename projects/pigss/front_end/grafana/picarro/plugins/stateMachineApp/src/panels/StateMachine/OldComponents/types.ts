// import { any } from "prop-types";

// export interface ButtonInfo {
//   caption: string;
//   className: string;
//   response: string;
// }

// export interface CommandPanelOptions {
//   uistatus: {
//     [key: string]: string;
//   };
//   ws_sender: (o: object) => void;
//   plan: Plan;
//   updatePanel: (x: number) => void;
//   layoutSwitch: () => {}
// }

// export interface ModalInfo {
//   show: boolean;
//   html: string;
//   num_buttons: number;
//   buttons: { [key: string]: ButtonInfo };
// }

// export interface Plan {
//   max_steps: number;
//   panel_to_show: number;
//   current_step: number;
//   focus: PlanFocus;
//   last_step: number;
//   steps: { [key: string]: PlanStep };
//   num_plan_files: number;
//   plan_files: { [key: string]: string };
//   plan_filename: string;
//   bank_names: {
//     [key: number]: {
//       name: string;
//       channels: { [key: number]: string };
//     };
//   };
// }

// export interface PlanFocus {
//   row: number;
//   column: number;
// }

// export interface PlanLoadPanelOptions {
//   plan: Plan;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   updatePanel: (x: number) => void;
//   loadPlan: (x: string) => void;
//   deleteFile: (x: string) => void;
//   getPlanFileNames: () => void;
//   fileNames: {[key: string]: string}
//   cancelLoadPlan: () => void;
// }

// export interface PlanLoadPanelEditOptions {
//   plan: Plan;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   getPlanFromFileName: (s: string) => void;
//   editPlan: () => void;
//   updatePanel: (x: number) => void;
//   getPlanFileNames: () => void;
//   fileNames: {[key: string]: string}
// }

// export interface PlanPanelOptions {
//   uistatus: { [key: string]: string };
//   plan: Plan;
//   setFocus: (row: number, column: number) => void;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   // addChanneltoPlan: (bank: number, channel: number) => void;
//   bankAddition: { [key: string]: number };
//   fileName: string;
//   loadFile: () => void;
//   savePlan: () => void;
//   updatePanel: (x: number) => void;
// }

// export interface PlanPanelLayoutOptions {
//   uistatus: { [key: string]: string };
//   plan: Plan;
//   setFocus: (row: number, column: number) => void;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   // addChanneltoPlan: (bank: number, channel: number) => void;
//   bankAddition: { [key: string]: number };
//   updatePanel: (x: number) => void;            
//   getPlanFileNames: () => void;
//   fileNames: {[key: string]: string}
// }

// export enum PlanPanelTypes {
//   NONE = 0,
//   PLAN = 1,
//   LOAD = 2,
//   EDIT = 3,
//   PREVIEW = 4
// }

// export interface PlanSavePanelOptions {
//   plan: Plan;
//   ws_sender: (o: object) => void;
//   isChanged: boolean;
//   updateFileName: (x: boolean) => void;
//   planSaved: (f: string) => void;
//   editPlan: () => void;
//   updatePanel: (x: number) => void;
//   fileNames: {[key: string]: string}
// }

// export interface BankConfig {
//   chan_mask: number;
//   clean: number;
// }

// export interface PlanStep {
//   banks: { [key: number]: BankConfig };
//   reference: number;
//   duration: number;
// }

// export interface EditPanelOptions {
//   uistatus: {
//     [key: string]: any;
//   };
//   ws_sender: (o: object) => void;
//   plan: Plan;
//   updatePanel: (x: number) => void;

// }
