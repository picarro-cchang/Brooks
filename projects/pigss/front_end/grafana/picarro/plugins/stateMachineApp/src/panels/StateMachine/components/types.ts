export interface PlanFocus {
    row: number;
    column: number;
}

export interface BankConfig {
    chan_mask: number;
    clean: number;
}

export interface PlanStep {
    banks: { [key: number]: BankConfig };
    reference: number;
    duration: number;
}

export interface Plan {
    max_steps: number;
    panel_to_show: number;
    current_step: number;
    focus: PlanFocus;
    last_step: number;
    steps: { [key: string]: PlanStep };
    num_plan_files: number;
    plan_files: { [key: string]: string };
    plan_filename: string;
    bank_names: {
      [key: number]: {
        name: string;
        channels: { [key: number]: string };
      };
    };
}

export interface PlanLayoutProps {
    uistatus: {
      [key: string]: any;
    };
    // ws_sender: (o: object) => void;
    plan: Plan;
    layoutSwitch: () => void;
    fileNames: {}
    getPlanFileNames: () => void;
    loadedFileName: string;
}

export interface RunLayoutProps {
    uistatus: {
      [key: string]: any;
    };
    ws_sender: (o: object) => void;
    plan: Plan;
    layoutSwitch: () => void;
    fileNames: {}
    loadedFileName: string;
    getPlanFileNames: () => void;
}

export enum PanelTypes {
    NONE = 0,
    LOAD = 1,
    EDIT = 2,
    PREVIEW = 3
}

export enum PlanPanelTypes {
    PLAN = 0,
    LOAD = 1,
    SAVE = 2,
}

export interface BankPanelPlanOptions {
    bank: number;
    uistatus: {
      bank?: { [bankNum: string]: string };
      clean?: { [bankNum: string]: string };
      channel?: { [bankNum: string]: { [channelNum: string]: string } };
    };
    // ws_sender: (o: object) => void;
    plan: Plan;
    addChanneltoPlan: (bank: number, channel: number) => void;
}

export interface LoadPanelCommandOptions {
    plan: Plan;
    ws_sender: (o: object) => void;
    updatePanel: (x: number) => void;
    // loadPlan: (x: string) => void;
    // deleteFile: (x: string) => void;
    fileNames: {}
    cancelLoadPlan: () => void;
    getPlanFromFileName: (fileName: string) => void;
    deleteFile: (file: string) => void;
    loadedFileName: string;
    planID: String;
  }

  export interface PreviewPanelOptions {
    plan: Plan;
    ws_sender: (o: object) => void;
    updatePanel: (x: number) => void;
    // loadPlan: (x: string) => void;
    // deleteFile: (x: string) => void;
    fileNames: {}
    cancelLoadPlan: () => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
    loadedFileName: string;
    planID: String;
  }


  export interface LoadPanelOptions {
    plan: Plan;
    // ws_sender: (o: object) => void;
    isEdited: boolean;
    updateFileName: (x: boolean) => void;
    updatePanel: (x: number) => void;
    fileNames: {}
    getPlanFromFileName: (fileName: string) => void;
    deleteFile: (file: string) => void;
    loadedFileName: string;
  }

export interface PlanPanelOptions {
    uistatus: { [key: string]: any };
    plan: Plan;
    isEdited: boolean;
    updateFileName: (x: boolean) => void;
    bankAddition: { [key: string]: number };
    fileName: string;
    updatePanel: (x: number) => void;
    layoutSwitch: () => void;
    planOverwrite: (s: object, x: String) => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
    updateSavedFileState: (plan: Plan) => void;
    planID: String;
  }

  export interface PlanSavePanelOptions {
    plan: Plan;
    // ws_sender: (o: object) => void;
    isEdited: boolean;
    updateFileName: (x: boolean) => void;
    planSaved: (f: string, d: object) => void;
    updatePanel: (x: number) => void;
    fileNames: {}
    deleteFile: (file: string) => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
  }

  export interface EditPanelOptions {
    uistatus: {
      [key: string]: any;
    };
    ws_sender: (o: object) => void;
    plan: Plan;
    updatePanel: (x: number) => void;
    planID: String;
  
  }
  export interface CommandPanelOptions {
    uistatus: {
      [key: string]: string;
    };
    ws_sender: (o: object) => void;
    plan: Plan;
    updatePanel: (x: number) => void;
    layoutSwitch: () => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
    loadedFileName: string;
    planID: String;
  }
