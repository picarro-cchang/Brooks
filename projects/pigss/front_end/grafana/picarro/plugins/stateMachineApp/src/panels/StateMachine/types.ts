import { any } from "prop-types";

export interface ButtonInfo {
  caption: string;
  className: string;
  response: string;
}

export interface CommandPanelOptions {
  uistatus: {
    [key: string]: string;
  };
  ws_sender: (o: object) => void;
  plan: Plan;
}

export interface ModalInfo {
  show: boolean;
  html: string;
  num_buttons: number;
  buttons: { [key: string]: ButtonInfo };
}

export interface Plan {
  max_steps: number;
  panel_to_show: number;
  current_step: number;
  focus: PlanFocus;
  last_step: number;
  steps: { [key: string]: PlanStep };
  num_plan_files: number;
  plan_files: { [key: string]: string };
  plan_filename: string;
  bank_names: {
    [key: number]: {
      name: string;
      channels: { [key: number]: string };
    };
  };
}

export interface PlanFocus {
  row: number;
  column: number;
}

export interface PlanLoadPanelOptions {
  plan: Plan;
  ws_sender: (o: object) => void;
  isChanged: boolean;
  updateFileName: (x: boolean) => void;
}

export interface PlanPanelOptions {
  uistatus: { [key: string]: string };
  plan: Plan;
  setFocus: (row: number, column: number) => void;
  ws_sender: (o: object) => void;
  isChanged: boolean;
  updateFileName: (x: boolean) => void;
  // addChanneltoPlan: (bank: number, channel: number) => void;
  bankAddition: {[key: string]: number}
}

export enum PlanPanelTypes {
  NONE = 0,
  PLAN = 1,
  LOAD = 2,
  SAVE = 3,
  EDIT = 4
}

export interface PlanSavePanelOptions {
  plan: Plan;
  ws_sender: (o: object) => void;
  isChanged: boolean;
  updateFileName: (x: boolean) => void;
}

export interface BankConfig {
  chan_mask: number;
  clean: number;
}

export interface PlanStep {
  banks: { [key: number]: BankConfig };
  reference: number;
  duration: number;
}

export interface EditPanelOptions {
  uistatus: {
    [key: string]: any;
  };
  ws_sender: (o: object) => void;
  plan: Plan;
}
