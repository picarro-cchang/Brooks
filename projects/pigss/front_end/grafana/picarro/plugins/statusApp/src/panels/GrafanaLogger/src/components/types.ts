import { GrafanaTheme } from '@grafana/ui';
import { TimeRange } from '@grafana/data';

export interface LogProps {
  level: object[];
  limit: number;
  timeRange: TimeRange;
  data: string[][];
}

export interface LogSectionProps {
  options: LogProps;
  theme: GrafanaTheme;
}

export interface CellProps {
  row: string[];
}
