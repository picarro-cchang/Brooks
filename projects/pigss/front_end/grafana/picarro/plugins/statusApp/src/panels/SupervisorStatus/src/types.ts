import { GrafanaTheme } from '@grafana/ui';

export interface SupervisorStatusProps {
}

export interface SupervisorStatusLayoutProps {
  theme: GrafanaTheme;
  options: SupervisorStatusProps;
}