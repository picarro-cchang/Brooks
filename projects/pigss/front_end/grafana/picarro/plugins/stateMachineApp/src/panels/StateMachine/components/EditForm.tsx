import React, { Component, PureComponent, ReactText } from "react";
import { EditPanelOptions } from "../types";

export class EditForm extends Component<EditPanelOptions> {
  state = {
    initialized: false,
    modal_info: {
      show: false,
      html: "",
      num_buttons: 0,
      buttons: {}
    },
    uistatus: {},
    plan: {
      bank_names: this.props.plan.bank_names
    }
  };
  banks: any;
  bank_list: any;

  render() {
    let edit_list = [];
    this.bank_list = [];
    this.banks = this.props.uistatus.bank;
    for (let num in this.banks) {
      let value = this.banks[num];
      if (value === "READY") {
        this.bank_list.push(num);
        edit_list.push(
          <div className="gf-form-group">
            <div className="row">
              <label className="edit-label"> Bank {num}</label>
              <input
                name={"bank" + num}
                className="col-sm-6 edit-input"
                type="text"
                defaultValue={this.state.plan.bank_names[num].name}
                maxLength={14}
              />
            </div>
            <div className="row">
              <label className="edit-label"> Channel 1 </label>
              <input
                name={"bank" + num + "1"}
                className="col-sm-6 edit-input"
                type="text"
                defaultValue={this.state.plan.bank_names[num].channels[1]}
                maxLength={8}
              />
            </div>
            <div className="row">
              <label className="edit-label"> Channel 2 </label>
              <input
                name={"bank" + num + "2"}
                className="col-sm-6 edit-input"
                type="text"
                defaultValue={this.state.plan.bank_names[num].channels[2]}
                maxLength={8}
              />
            </div>
            <div className="row">
              <label className="edit-label"> Channel 3 </label>
              <input
                name={"bank" + num + "3"}
                className="col-sm-6 edit-input"
                type="text"
                defaultValue={this.state.plan.bank_names[num].channels[3]}
                maxLength={8}
              />
            </div>
            <div className="row">
              <label className="edit-label"> Channel 4 </label>
              <input
                name={"bank" + num + "4"}
                className="col-sm-6 edit-input"
                type="text"
                defaultValue={this.state.plan.bank_names[num].channels[4]}
                maxLength={8}
              />
            </div>
            <div className="row">
              <label className="edit-label"> Channel 5 </label>
              <input
                name={"bank" + num + "5"}
                className="col-sm-6 edit-input"
                type="text"
                defaultValue={this.state.plan.bank_names[num].channels[5]}
                maxLength={8}
              />
            </div>
            <div className="row">
              <label className="edit-label"> Channel 6 </label>
              <input
                name={"bank" + num + "6"}
                className="col-sm-6 edit-input"
                type="text"
                defaultValue={this.state.plan.bank_names[num].channels[6]}
                maxLength={8}
              />
            </div>
            <div className="row">
              <label className="edit-label"> Channel 7 </label>
              <input
                name={"bank" + num + "7"}
                className="col-sm-6 edit-input"
                type="text"
                defaultValue={this.state.plan.bank_names[num].channels[7]}
                maxLength={8}
              />
            </div>
            <div className="row">
              <label className="edit-label"> Channel 8 </label>
              <input
                name={"bank" + num + "8"}
                className="col-sm-6 edit-input"
                type="text"
                defaultValue={this.state.plan.bank_names[num].channels[8]}
                maxLength={8}
              />
            </div>
          </div>
        );
      }
    }
    return <div className="panel-plan-inner">{edit_list}</div>;
  }
}
