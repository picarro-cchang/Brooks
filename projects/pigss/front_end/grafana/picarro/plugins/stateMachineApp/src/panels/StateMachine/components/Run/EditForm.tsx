import React, { Component, PureComponent, ReactText } from "react";
import { EditPanelOptions } from "../types";

class EditForm extends Component<any, any> {
  state = {
    plan: {
      bank_names: this.props.plan.bank_names
    }
  };
  banks: any;
  bank_list: any;

  render() {
    const edit_list = [];
    this.bank_list = [];
    this.banks = this.props.uistatus.bank;
    for (const num in this.banks) {
      const value = this.banks[num];
      if (value === "READY") {
        this.bank_list.push(num);
        edit_list.push(<div key={"bank" + num} className="row">
            <label className="edit-label"> Bank {num}</label>
      </div>)
        for (let i = 1; i <= 8; i++) {
          let portNumber = (Number(num) - 1) * 8 + i
          edit_list.push(
            <div key={portNumber} className="row">
              <label className="edit-label"> Port {portNumber}: </label>
              <input                
                name={"bank" + num + i}
                className="col-sm-6 edit-input"
                type="text"
                onChange={event =>
                  this.props.handleChannelNameChange(
                    event.target.value,
                    num.toString(),
                    i
                  )
                }
                value={this.props.plan.bank_names[num].channels[i]}
                maxLength={15}
              />
            </div>
          );
        }
      }
    }
    return (
      <div className="panel-plan-inner">
        <div className="gf-form-group">
          {edit_list}
        </div>
      </div>);
  }
}

export default EditForm;
