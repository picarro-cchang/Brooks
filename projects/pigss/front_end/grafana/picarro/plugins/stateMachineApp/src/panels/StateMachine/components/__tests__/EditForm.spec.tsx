import React from 'react';
import { shallow, mount } from 'enzyme';
import 'jest-styled-components';
import WS from "jest-websocket-mock";
import { EditPanelOptions } from '../../types';
import EditForm from '../EditForm';

const mockClick = jest.fn((element) => {return element;});

const defaultProps: EditPanelOptions = {
    uistatus: {
        "bank": {
            "1" : "READY",
            "2" : "READY",
            "3" : "DISABLED"
        }
    },
    ws_sender: mockClick,
    plan: {
        "max_steps": 32,
        "panel_to_show": 0,
        "current_step": 8,
        "focus": {
            "row": 15,
            "column": 1
        },
        "last_step": 14,
        "steps": {
            "1": {
            "banks": {
                "1": {
                "clean": 0,
                "chan_mask": 1
                },
                "2": {
                "clean": 0,
                "chan_mask": 0
                }
            },
            "reference": 0,
            "duration": 30
            },
            "2": {
            "banks": {
                "1": {
                "clean": 0,
                "chan_mask": 2
                },
                "2": {
                "clean": 0,
                "chan_mask": 0
                }
            },
            "reference": 0,
            "duration": 30
            },
            "3": {
            "banks": {
                "1": {
                "clean": 0,
                "chan_mask": 4
                },
                "2": {
                "clean": 0,
                "chan_mask": 0
                }
            },
            "reference": 0,
            "duration": 30
            },
            "4": {
            "banks": {
                "1": {
                "clean": 0,
                "chan_mask": 8
                },
                "2": {
                "clean": 0,
                "chan_mask": 0
                }
            },
            "reference": 0,
            "duration": 30
            },
            "5": {
            "banks": {
                "1": {
                "clean": 0,
                "chan_mask": 16
                },
                "2": {
                "clean": 0,
                "chan_mask": 0
                }
            },
            "reference": 0,
            "duration": 30
            },
            "6": {
            "banks": {
                "1": {
                "clean": 0,
                "chan_mask": 32
                },
                "2": {
                "clean": 0,
                "chan_mask": 0
                }
            },
            "reference": 0,
            "duration": 30
            },
            "7": {
            "banks": {
                "1": {
                "clean": 0,
                "chan_mask": 64
                },
                "2": {
                "clean": 0,
                "chan_mask": 0
                }
            },
            "reference": 0,
            "duration": 30
            },
            "8": {
            "banks": {
                "1": {
                "clean": 0,
                "chan_mask": 128
                },
                "2": {
                "clean": 0,
                "chan_mask": 0
                }
            },
            "reference": 0,
            "duration": 30
            },
            "9": {
            "banks": {
                "1": {
                "clean": 1,
                "chan_mask": 0
                },
                "2": {
                "clean": 0,
                "chan_mask": 0
                }
            },
            "reference": 0,
            "duration": 30
            },
            "10": {
            "banks": {
                "1": {
                "clean": 0,
                "chan_mask": 0
                },
                "2": {
                "clean": 0,
                "chan_mask": 0
                }
            },
            "reference": 1,
            "duration": 30
            },
            "11": {
            "banks": {
                "1": {
                "clean": 0,
                "chan_mask": 0
                },
                "2": {
                "clean": 0,
                "chan_mask": 1
                }
            },
            "reference": 0,
            "duration": 30
            },
            "12": {
            "banks": {
                "1": {
                "clean": 0,
                "chan_mask": 0
                },
                "2": {
                "clean": 0,
                "chan_mask": 2
                }
            },
            "reference": 0,
            "duration": 30
            },
            "13": {
            "banks": {
                "1": {
                "clean": 0,
                "chan_mask": 0
                },
                "2": {
                "clean": 0,
                "chan_mask": 4
                }
            },
            "reference": 0,
            "duration": 30
            },
            "14": {
            "banks": {
                "1": {
                "clean": 0,
                "chan_mask": 0
                },
                "2": {
                "clean": 0,
                "chan_mask": 8
                }
            },
            "reference": 0,
            "duration": 30
            }
        },
        "num_plan_files": 7,
        "plan_files": {
            "1": "1234567890123456789012345678",
            "2": "30_seconds",
            "3": "__default__",
            "4": "abcdefghijklmnopqrstwxyzabc",
            "5": "abcdefghijklmnopqrstwxyzabcd",
            "6": "kp_test",
            "7": "kp_test_2"
        },
        "plan_filename": "__default__",
        "bank_names": {
            "1": {
            "name": "B 1",
            "channels": {
                "1": "Channel 1",
                "2": "Channel 2",
                "3": "Ch. 3",
                "4": "Ch. 4",
                "5": "Ch. 5",
                "6": "Ch. 6",
                "7": "Ch. 7",
                "8": "Ch. 8"
            }
            },
            "2": {
            "name": "Bank 2",
            "channels": {
                "1": "Ch. 1",
                "2": "Ch. 2",
                "3": "Ch. 3",
                "4": "Ch. 4",
                "5": "Ch. 5",
                "6": "Ch. 6",
                "7": "Ch. 7",
                "8": "Ch. 8"
            }
            },
            "3": {
            "name": "Bank 3",
            "channels": {
                "1": "Ch. 1",
                "2": "Ch. 2",
                "3": "Ch. 3",
                "4": "Ch. 4",
                "5": "Ch. 5",
                "6": "Ch. 6",
                "7": "Ch. 7",
                "8": "Ch. 8"
            }
            },
        },
    }
};


describe('<EditForm />', () => {
  const part = mount(<EditForm {...defaultProps} />);
  const wrapper = shallow(<EditForm {...defaultProps} />);

  it('Renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Renders with correct bank/channel Names', () => {
      const value = wrapper.find('input[name="bank1"]').props().value;
      const chn2value = wrapper.find('input[name="bank12"]').props().value;

      expect(value).toEqual('B 1');
      expect(chn2value).toEqual('Channel 2');

  });

});

