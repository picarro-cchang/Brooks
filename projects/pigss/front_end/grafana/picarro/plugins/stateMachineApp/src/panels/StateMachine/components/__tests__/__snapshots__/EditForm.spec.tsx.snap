// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditForm /> Renders correctly 1`] = `
<div
  className="panel-plan-inner"
>
  <div
    className="gf-form-group"
  >
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Bank 
        1
      </label>
      <input
        className="col-sm-6 edit-input"
        defaultValue="Loser 1"
        maxLength={14}
        name="bank1"
        type="text"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 1 
      </label>
      <input
        className="col-sm-6 edit-input"
        defaultValue="Channel 1"
        maxLength={8}
        name="bank11"
        type="text"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 2 
      </label>
      <input
        className="col-sm-6 edit-input"
        defaultValue="Channel 2"
        maxLength={8}
        name="bank12"
        type="text"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 3 
      </label>
      <input
        className="col-sm-6 edit-input"
        defaultValue="Ch. 3"
        maxLength={8}
        name="bank13"
        type="text"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 4 
      </label>
      <input
        className="col-sm-6 edit-input"
        defaultValue="Ch. 4"
        maxLength={8}
        name="bank14"
        type="text"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 5 
      </label>
      <input
        className="col-sm-6 edit-input"
        defaultValue="Ch. 5"
        maxLength={8}
        name="bank15"
        type="text"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 6 
      </label>
      <input
        className="col-sm-6 edit-input"
        defaultValue="Ch. 6"
        maxLength={8}
        name="bank16"
        type="text"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 7 
      </label>
      <input
        className="col-sm-6 edit-input"
        defaultValue="Ch. 7"
        maxLength={8}
        name="bank17"
        type="text"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 8 
      </label>
      <input
        className="col-sm-6 edit-input"
        defaultValue="Ch. 8"
        maxLength={8}
        name="bank18"
        type="text"
      />
    </div>
  </div>
  <div
    className="gf-form-group"
  >
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Bank 
        2
      </label>
      <input
        className="col-sm-6 edit-input"
        defaultValue="Bank 2"
        maxLength={14}
        name="bank2"
        type="text"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 1 
      </label>
      <input
        className="col-sm-6 edit-input"
        defaultValue="Ch. 1"
        maxLength={8}
        name="bank21"
        type="text"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 2 
      </label>
      <input
        className="col-sm-6 edit-input"
        defaultValue="Ch. 2"
        maxLength={8}
        name="bank22"
        type="text"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 3 
      </label>
      <input
        className="col-sm-6 edit-input"
        defaultValue="Ch. 3"
        maxLength={8}
        name="bank23"
        type="text"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 4 
      </label>
      <input
        className="col-sm-6 edit-input"
        defaultValue="Ch. 4"
        maxLength={8}
        name="bank24"
        type="text"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 5 
      </label>
      <input
        className="col-sm-6 edit-input"
        defaultValue="Ch. 5"
        maxLength={8}
        name="bank25"
        type="text"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 6 
      </label>
      <input
        className="col-sm-6 edit-input"
        defaultValue="Ch. 6"
        maxLength={8}
        name="bank26"
        type="text"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 7 
      </label>
      <input
        className="col-sm-6 edit-input"
        defaultValue="Ch. 7"
        maxLength={8}
        name="bank27"
        type="text"
      />
    </div>
    <div
      className="row"
    >
      <label
        className="edit-label"
      >
         Channel 8 
      </label>
      <input
        className="col-sm-6 edit-input"
        defaultValue="Ch. 8"
        maxLength={8}
        name="bank28"
        type="text"
      />
    </div>
  </div>
</div>
`;
